### ISSUE TITLE
`dangerouslySetInnerHTML` is not working in `hono/jsx/dom`

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#1944]
### What version of Hono are you using?

4.0.0-rc.0

### What runtime/platform is your app running on?

any

### What steps can reproduce the bug?

```tsx
import { render } from 'hono/jsx/dom'

const root = document.getElementById('root') as HTMLElement

const Div = () => {
  const inner = { __html: '<h1>Hello</h1>' }
  return <div dangerouslySetInnerHTML={inner} />
}

render(<Div />, root)
```

### What is the expected behavior?

Render the content:

```html
<div><h1>Hello</h1></div>
```

### What do you see instead?

```html
<div dangerouslysetinnerhtml="[object Object]"></div>
```
<img width="829" alt="Screenshot 2024-01-11 at 23 27 06" src="https://github.com/honojs/hono/assets/10682/03e38379-066b-4576-b066-fe5f3fa455fc">

**Comment by EdamAme-x** [comment]
https://github.com/EdamAme-x/hono/blob/5ba31090dabe4dd275babde22cfb427167ee4add/src/jsx/dom/index.ts#L348

It looks like it could be implemented here.

**Comment by yusukebe** [comment]
@EdamAme-x 

Thanks. But this will be fixed by #1955.

**Comment by yusukebe** [comment]
Fixed, thanks!
